### ISSUE TITLE
Reverse/DoubleEnded EnumIter

### ISSUE THREAD

**Issue by Nevsor** [Peternator7/strum#59]
It would be nice to be able to iterate over enums in reverse order.

Currently I have to use the workaround `my_enum::iter().collect::<Vec<_>>().iter().rev()`, which is quite ugly and probably less performant.

This could be implemented by making the IntoEnumIter::iter function return a DoubleEndedIterator to allow code like this:

``` rust
#[derive(EnumIter)]
enum Colors {Red, Green, Blue}

use strum::IntoEnumIterator;
for color in Colors::iter().rev() {
    // Do something with color.
}
```

This approach might incur a runtime cost, because we would have to add another field to the iterator struct (a second index for `next_back` similar to the existing `idx`) even if the user does not call `next_back`.

Another possible way to implement this is to add a new method, that returns a reverse or double-ended iterator.

Are you interested in adding this functionality?

**Comment by Peternator7** [comment]
Thanks for the feedback! I think it's a very useful feature :) I'd be inclined to add a field to the existing iterator structure so as to keep the api surface small.

I'm unfortunately too busy to get to this in the short term, but if you open a PR, i'd be more than happy to review it.

**Comment by Peternator7** [comment]
Thanks for the PR :) I'll get this published in `0.17`
